planner/core/expression_rewriter_test.go

```diff
@@ -130,6 +130,21 @@ func (s *testExpressionRewriterSuite) TestDefaultFunction(c *C) {
 
 	tk.MustExec("update t1 set c = c + default(c)")
 	tk.MustQuery("select c from t1").Check(testkit.Rows("11"))
+
+	tk.MustExec("create table t6(a int default -1, b int)")
+	tk.MustExec(`insert into t6 values (0, 0), (1, 1), (2, 2)`)
+	tk.MustExec("create table t7(a int default 1, b int)")
+	tk.MustExec(`insert into t7 values (0, 0), (1, 1), (2, 2)`)
+
+	tk.MustQuery(`select a from t6 where a > (select default(a) from t7 where t6.a = t7.a)`).Check(testkit.Rows("2"))
+	tk.MustQuery(`select a, default(a) from t6 where a > (select default(a) from t7 where t6.a = t7.a)`).Check(testkit.Rows("2 -1"))
+
+	tk.MustExec("create table t8(a int default 1, b int default -1)")
+	tk.MustExec(`insert into t8 values (0, 0), (1, 1)`)
+
+	tk.MustQuery(`select a, a from t8 order by default(a)`).Check(testkit.Rows("0 0", "1 1"))
+	tk.MustQuery(`select a from t8 order by default(b)`).Check(testkit.Rows("0", "1"))
+	tk.MustQuery(`select a from t8 order by default(b) * a`).Check(testkit.Rows("1", "0"))
```

Comment:
Would you please add some test that is invalid in resolving column names?

Reply:
It exists at
https://github.com/pingcap/tidb/pull/19709/files#diff-1e69525b01b51d983863da3fe1d4a21fR93